diff --git a/area_detector_handlers/handlers.py b/area_detector_handlers/handlers.py
index 804a447..c619007 100644
--- a/area_detector_handlers/handlers.py
+++ b/area_detector_handlers/handlers.py
@@ -126,7 +126,14 @@ class HDF5DatasetSliceHandlerPureNumpy(HandlerBase):
 
     def close(self):
         super().close()
-        self._file.close()
+        # Intentionally do not close the file. Typically we would have
+        # self._file.close() here to free toe resources associated with the
+        # open file. However, libhdf5 keeps its own account of which files are
+        # open, and if we explicitly close a file like that, *all* handles on
+        # it will become invalid. Thus, any other handler instances using this
+        # file would be broken by us, and indeed we discovered the hard way.
+        # Therefore, we simply drop the reference to the file hanlder here and
+        # let libhdf5 take care of the cleanup.
         self._file = None
 
 
diff --git a/area_detector_handlers/tests/conftest.py b/area_detector_handlers/tests/conftest.py
index c4b0a64..fda3c36 100644
--- a/area_detector_handlers/tests/conftest.py
+++ b/area_detector_handlers/tests/conftest.py
@@ -65,7 +65,7 @@ def hdf5_files(request):
 
     data = np.concatenate([
         np.ones((fpp, N_rows, N_cols)) * pt for pt in range(N_points)])
-    with h5py.File(full_path) as file:
+    with h5py.File(full_path, 'w') as file:
         file.create_dataset('entry/data/data', data=data)
 
     def finalize():
